src/SongList.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState } from "react";
+import { useHandleFetchAndLoad } from "./useHandleFetchAndLoad";
+
+export const Songlist = (props: { playlistId: string }): React.ReactElement => {
+  const [songList, setSongList] = useState({ index: 0, tracks: [] });
+  const endpoint = `https://api.spotify.com/v1/playlists/${props.playlistId}/tracks?offset=${songList.index}`;
```

Comment:
you should definitely move the basurl and base strings as a constant value that way you can change it in one place if this ends up moving around

Reply:
the other benefit is you can append things using a url encoder so you ensure that they are url friendly.

Reply:
Agreed. Realistically, all interaction with the spotify api should be done through a new set of methods that deal specifically with interacting with spotify via rest (essentially making an sdk). 
We can work on that later though.

Reply:
Another benefit of making an API here is that it would be a natural place to define all the spotify types you would expect, that way you could export these types from one place (there are a few places where these types are duplicated).